repository/spark/docs/latest/history-server.md

```diff
@@ -5,30 +5,75 @@ Spark History Server Configuration
 
 Required software:
 * K8s cluster
-* [KUDO CLI Plugin](https://kudo.dev/docs/#install-kudo-cli) 0.8.0 or higher
+* [KUDO CLI Plugin](https://kudo.dev/docs/#install-kudo-cli) 0.10.1 or higher
 
 ## Installing Spark Operator with History Server Enabled
 
 ```bash
 kubectl kudo install spark --instance=spark-operator \
     -p enableHistoryServer=true \
-    -p historyServerFsLogDirectory="s3a://<BUCKET_NAME>/<FOLDER>" \
-    -p historyServerOpts="-Dspark.hadoop.fs.s3a.access.key=<AWS_ACCESS_KEY_ID> 
-        -Dspark.hadoop.fs.s3a.secret.key=<AWS_SECRET_ACCESS_KEY>
-        -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"
+    -p historyServerFsLogDirectory="<log directory>" \
 ```
+This will deploy a Pod and Service for the `Spark History Server` with the `Spark Event Log` directory configured via the `historyServerFsLogDirectory` parameter. Spark Operator also supports Persistent Volume Claim (PVC) based storage. There is a parameter `historyServerPVCName` to pass the name of the PVC. Make sure that provided PVC should have `ReadWriteMany` [access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) supported.
 
-This will deploy a Pod and Service for the `Spark History Server` with the `Spark Event Log` directory configured via the `historyServerFsLogDirectory` parameter. This is an S3 backed storage for event logs. Spark Operator also supports Persistent Volume Claim (PVC) based storage. There is a parameter `historyServerPVCName` to pass the name of the PVC. Make sure that provided PVC should have `ReadWriteMany` [access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) supported.
+If you want to write and read event logs to/from an S3 bucket, you can do the following:
+1) Create a `Secret` which will contain Spark configuration (file name is important):
+```bash
+$ cat << 'EOF' >> spark-defaults.conf
+spark.hadoop.fs.s3a.access.key <AWS_ACCESS_KEY_ID>
+spark.hadoop.fs.s3a.secret.key <AWS_SECRET_ACCESS_KEY>
+spark.hadoop.fs.s3a.impl org.apache.hadoop.fs.s3a.S3AFileSystem
+EOF
+$ kubectl create secret generic spark-conf --from-file spark-defaults.conf --namespace spark
+```
+
+2) Install Spark Operator with the following parameters:
+```bash
+kubectl kudo install spark --instance=spark-operator \
+    -p enableHistoryServer=true \
+    -p historyServerFsLogDirectory="s3a://<BUCKET_NAME>/<FOLDER>" \
+    -p historyServerSparkConfSecret=spark-conf
+```
 
 ## Creating Spark Application
 
-Make sure [`specs/spark-application.yaml`](../../../../../specs/spark-application.yaml) has these properties specified under `spec.sparkConf`:
+When submitting Spark Applications to the Operator with Spark History server enabled, you need to provide additional 
+configuration depending on storage type you are using for event logging. 
```

Comment:
I'd suggest avoiding appeals which use personal pronouns with verbs like "need" or "must" in the documentation. How about we use something like that?
```
When Spark History server is enabled, an additional configuration such as security credentials can be required for accessing the storage backend being used for event logging.
```

Reply:
yes, looks good, updated